### PR TITLE
Add contributing doc for IDE code style analyzer documentation

### PR DESCRIPTION
**Pull request by mavasani**

### CODE REVIEW

File: docs/contributing/Documentation for IDE CodeStyle analyzers.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Documentation for IDE CodeStyle analyzers
+
+## Overview
+
+1. Official documentation for all [IDE analyzer diagnostic IDs](../../src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs) is added to `dotnet/docs` repo at https://github.com/dotnet/docs/tree/master/docs/fundamentals/code-analysis/style-rules
```
Comment (Youssef1313) [inline docs/contributing/Documentation for IDE CodeStyle analyzers.md:5]: Some markdown parsers doesn't make link text clickable by default. See [MD034](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md034---bare-url-used).

```suggestion
1. Official documentation for all [IDE analyzer diagnostic IDs](../../src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs) is added to `dotnet/docs` repo at <https://github.com/dotnet/docs/tree/master/docs/fundamentals/code-analysis/style-rules>
```

File: docs/contributing/Documentation for IDE CodeStyle analyzers.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+2. Each IDE diagnostic ID has a dedicated documentation page. For example:
+   1. Diagnostic with code style option: https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0011
```
Comment (Youssef1313) [inline docs/contributing/Documentation for IDE CodeStyle analyzers.md:8]:
```suggestion
   1. Diagnostic with code style option: <https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0011>
```

File: docs/contributing/Documentation for IDE CodeStyle analyzers.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+   2. Diagnostic without code style option: https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0004
```
Comment (Youssef1313) [inline docs/contributing/Documentation for IDE CodeStyle analyzers.md:9]:
```suggestion
   2. Diagnostic without code style option: <https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0004>
```

File: docs/contributing/Documentation for IDE CodeStyle analyzers.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+   3. Multiple diagnostic IDs sharing the same option(s): https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0003-ide0009
```
Comment (Youssef1313) [inline docs/contributing/Documentation for IDE CodeStyle analyzers.md:10]:
```suggestion
   3. Multiple diagnostic IDs sharing the same option(s): <https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0003-ide0009>
```

File: docs/contributing/Documentation for IDE CodeStyle analyzers.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+3. We have tabular indices for IDE diagnostic IDs and code style options for easier reference and navigation:
+   1. Primary index (rule ID + code style options): https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/
```
Comment (Youssef1313) [inline docs/contributing/Documentation for IDE CodeStyle analyzers.md:13]:
```suggestion
   1. Primary index (rule ID + code style options): <https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/>
```

File: docs/contributing/Documentation for IDE CodeStyle analyzers.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+   2. Secondary index (code style options only): https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/language-rules#net-style-rules
```
Comment (Youssef1313) [inline docs/contributing/Documentation for IDE CodeStyle analyzers.md:14]:
```suggestion
   2. Secondary index (code style options only): <https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/language-rules#net-style-rules>
```

File: docs/contributing/Documentation for IDE CodeStyle analyzers.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+   3. Indices for each preference/option group. For example:
+      1. Modifier preferences: https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/modifier-preferences
```
Comment (Youssef1313) [inline docs/contributing/Documentation for IDE CodeStyle analyzers.md:16]:
```suggestion
      1. Modifier preferences: <https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/modifier-preferences>
```

File: docs/contributing/Documentation for IDE CodeStyle analyzers.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      2. Expression-level preferences: https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/expression-level-preferences
```
Comment (Youssef1313) [inline docs/contributing/Documentation for IDE CodeStyle analyzers.md:17]:
```suggestion
      2. Expression-level preferences: <https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/expression-level-preferences>
```

File: .github/ISSUE_TEMPLATE/analyzer-suggestion.md [hunk 9]
```diff
@@ -19,3 +19,7 @@ A small code snippet that describes a case that the analyzer should report.
 **Additional information:**
 
 Any more additional information you would like to add.
+
+**Documentation requirements:**
+
+When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](../../docs/contributing/Documentation for IDE CodeStyle analyzers.md)
```
Comment (JoeRobich) [inline .github/ISSUE_TEMPLATE/analyzer-suggestion.md:25]: Nit: In the case that it is a sentence.
```suggestion
When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](../../docs/contributing/Documentation for IDE CodeStyle analyzers.md).
```

File: docs/contributing/Documentation for IDE CodeStyle analyzers.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
+# Documentation for IDE CodeStyle analyzers
+
+## Overview
+
+1. Official documentation for all [IDE analyzer diagnostic IDs](../../src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs) is added to `dotnet/docs` repo at <https://github.com/dotnet/docs/tree/master/docs/fundamentals/code-analysis/style-rules>
```
Comment (JoeRobich) [inline docs/contributing/Documentation for IDE CodeStyle analyzers.md:5]: Nit: In the case that it is a sentence.
```suggestion
1. Official documentation for all [IDE analyzer diagnostic IDs](../../src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs) is added to `dotnet/docs` repo at <https://github.com/dotnet/docs/tree/master/docs/fundamentals/code-analysis/style-rules>.
```

**Review by JoeRobich — APPROVED** [review]
LGTM

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**